data-prepper-core/src/main/java/org/opensearch/dataprepper/sourcecoordination/DynamoDbSourceCoordinator.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.sourcecoordination;
+
+import org.opensearch.dataprepper.model.source.SourceCoordinator;
+import org.opensearch.dataprepper.model.source.SourcePartition;
+import org.opensearch.dataprepper.parser.model.sourcecoordination.SourceCoordinationConfig;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * An implementation of {@link org.opensearch.dataprepper.model.source.SourceCoordinator} when DynamoDB is used at the distributed store
+ * for source_coordination
+ * @since 2.2
+ */
+public class DynamoDbSourceCoordinator implements SourceCoordinator {
+
+    private final SourceCoordinationConfig sourceCoordinationConfig;
+
+    public DynamoDbSourceCoordinator(final SourceCoordinationConfig sourceCoordinationConfig) {
+        this.sourceCoordinationConfig = sourceCoordinationConfig;
+    }
+
+    @Override
+    public void createPartitions(List<String> partitionKeys) {
+
```

Comment:
Nit: TODO

Reply:
I can throw an `UnsupportedOperationException` as well instead if that makes sense